src/Models/Scopes/UserClientScope.php

```diff
@@ -13,25 +13,28 @@
 
 class UserClientScope implements Scope
 {
+    protected static array $clients = [];
+
     public function apply(Builder $builder, Model $model): void
     {
         // Don't apply scope if no user is authenticated
         if (! Auth::hasUser()) {
             return;
         }
 
-        $clients = ($user = Auth::user()) instanceof User ?
+        // write the clients to a static property to avoid multiple queries if the scope is used multiple times
+        static::$clients ??= ($user = Auth::user()) instanceof User ?
```

Comment:
Can't the user's clients be cached? They surely will not be updated that often. And even then could the cache be regenerated.